pkg/cmd/cli/secrets/addsecret_to_obj.go

```diff
@@ -0,0 +1,268 @@
+package secrets
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/meta"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	AttachSecretRecommendedName = "add"
+
+	attachSecretLong = `Add secrets to a ServiceAccount
+
+After you have created a secret, you probably want to make use of that secret inside of a pod, for a build, or as an image pull secret.  In order to do that, you must add your secret to a service account.
+
+To use your secret inside of a pod or as a push, pull, or source secret for a build, you must add a 'mount' secret to your service account like this:
+
+  $ %s serviceaccount/sa-name secrets/secret-name secrets/another-secret-name
+
+To use your secret as an image pull secret, you must add a 'pull' secret to your service account like this:
+
+  $ %s serviceaccount/sa-name secrets/secret-name --for=pull
+`
+)
+
+type SecretType string
+
+var (
+	PullType  SecretType = "pull"
+	MountType SecretType = "mount"
+)
+
+type AttachSecretOptions struct {
```

Comment:
> AddSecretToServiceAccountOptions ?

@smarterclayton wants to use the same command to handle adds to other target types.